docs/content/users/extend/additional-services.md

```diff
@@ -105,18 +105,31 @@ If you have add-ons that were installed before v1.22, update them with `ddev add
 
 ## Adding Custom Configuration to an Add-on
 
-Sometimes it's necessary to add custom configuration to an add-on. For example, in [`ddev-redis-7`](https://github.com/ddev/ddev-redis-7) the `image` is set to `image: redis:7.2-alpine`. If you wanted to change this to use `7.0-alpine` instead, you would have two choices:
+Sometimes it's necessary to add custom configuration to an add-on. For example, in [`ddev-redis`](https://github.com/ddev/ddev-redis) the `image` is set to `image: redis:${DDEV_REDIS_VERSION:-6-bullseye}`. If you wanted to change this to use `7.0-bookworm` instead, you would have two choices:
 
-1. Remove the `#ddev-generated` line in `docker-compose.redis-7.yaml` and edit the relevant line.
-2. Add a `.ddev/docker-compose.redis-7_extra.yaml` with the contents:
+1. With DDEV v1.23.5+, the override can be done using a special `.ddev/.env.*` file, where `*` is the name of the add-on (and its service), in this case `.ddev/.env.redis`:
 
-```yaml
-  services:
-    redis:
-      image: redis:7.0-alpine
-```
+    ```bash
+    ddev add-on get ddev/ddev-redis --environment="DDEV_REDIS_VERSION=7.0-alpine"
+    ddev restart
+    ```
+   
+    You can override `.ddev/.env.*` with `.ddev/.env.*.override` file in case you need more customization.
+
+    !!!warning "Environment variables cannot be overridden in some cases"
+        If your `docker-compose.*.yaml` file already has some variables in the `environment` or `env_file` stanza, you cannot override them with `.ddev/.env.*` or `.ddev/.env.*.override` files.
+        
+        If this is the case, move these variables to the appropriate `.ddev/.env.*` file to be able to override them with a `.ddev/.env.*.override` file.
```

Comment:
This is why I removed this section from the app template:

```yaml
    {{ if .EnvFile }}
    env_file: '{{ .EnvFile }}'
    {{ end }}
```

Otherwise, I couldn't override any existing variables from `.ddev/.env` with `.ddev/.env.override`.